### ISSUE TITLE
Coupling loss

### ISSUE THREAD

**Issue by ttgump** [AllenInstitute/coupledAE#2]
Hi,
I have a question about the coupling loss. According the the Eq 6 in your NIPS paper, the coupling loss is 
<img width="612" alt="Screen Shot 2020-01-16 at 1 59 37 PM" src="https://user-images.githubusercontent.com/1187979/72554338-7772fc00-3868-11ea-9eef-6ad3f4f487db.png">
In the line 50 of coupling_functions.py, it is
`L = mean_squared_error(zi, zj)/tf.math.minimum(minvar_i, minvar_j)`
According to Eq6, Should the loss be like this?
`zi_dash = (zi - tf.reduce_mean(zi, axis=0))/tf.math.reduce_variance(zi, axis=0)`
`zj_dash = (zj - tf.reduce_mean(zj, axis=0))/tf.math.reduce_variance(zj, axis=0)`
`L = mean_squared_error(zi_dash, zj_dash)/tf.math.minimum(minvar_i, minvar_j)`
Thanks!

**Comment by rhngla** [comment]
In our implementation conventional `BatchNorm` layers remove the mean in `zi` and `zj`, using which the coupling loss is calculated. 

Ideally we want population mean to be 0. The batch estimate of the population mean may be too variable from one batch to the next depending on batch size, and can adversely affect the alignment. Using a high momentum value in the `BatchNorm` helps us get around that potential issue. Your implementation (explicitly removing batch mean) could be sufficient in case your batch sizes are large enough. 

Thank you for pointing this out - I'll add this as a comment in the `coupling.py`.
